tests/unit/test_plugins.py

```diff
@@ -23,143 +24,125 @@ class PluginMock:  # pylint: disable=missing-class-docstring, too-few-public-met
+        request.remote_addr = '127.0.0.1'
+        request.headers = {'X-Forwarded-For': '1.2.3.4, 127.0.0.1', 'X-Real-IP': '127.0.0.1'}
         search = get_search_mock(query='ip', pageno=2)
-        store.call(store.plugins, 'post_search', request, search)
-        self.assertFalse('ip' in search.result_container.answers)
-
-        # User agent test
+        self.store.call(self.store.plugins, 'post_search', request, search)
+        self.assertNotIn('ip', search.result_container.answers)
+
+    @parameterized.expand(
+        [
+            'user-agent',
+            'What is my User-Agent?',
+        ]
+    )
+    def test_user_agent_in_answer(self, query: str):
         request = Mock(user_agent=Mock(string='Mock'))
-
-        search = get_search_mock(query='user-agent', pageno=1)
-        store.call(store.plugins, 'post_search', request, search)
-        self.assertTrue('Mock' in search.result_container.answers["user-agent"]["answer"])
-
-        search = get_search_mock(query='user-agent', pageno=2)
-        store.call(store.plugins, 'post_search', request, search)
-        self.assertFalse('user-agent' in search.result_container.answers)
-
-        search = get_search_mock(query='user-agent', pageno=1)
-        store.call(store.plugins, 'post_search', request, search)
-        self.assertTrue('Mock' in search.result_container.answers["user-agent"]["answer"])
-
-        search = get_search_mock(query='user-agent', pageno=2)
-        store.call(store.plugins, 'post_search', request, search)
-        self.assertFalse('user-agent' in search.result_container.answers)
-
-        search = get_search_mock(query='What is my User-Agent?', pageno=1)
-        store.call(store.plugins, 'post_search', request, search)
-        self.assertTrue('Mock' in search.result_container.answers["user-agent"]["answer"])
-
-        search = get_search_mock(query='What is my User-Agent?', pageno=2)
-        store.call(store.plugins, 'post_search', request, search)
-        self.assertFalse('user-agent' in search.result_container.answers)
+        search = get_search_mock(query=query, pageno=1)
+        self.store.call(self.store.plugins, 'post_search', request, search)
+        self.assertIn('Mock', search.result_container.answers["user-agent"]["answer"])
+
+    @parameterized.expand(
+        [
+            'user-agent',
+            'What is my User-Agent?',
+        ]
+    )
+    def test_user_agent_not_in_answer(self, query: str):
+        request = Mock(user_agent=Mock(string='Mock'))
+        search = get_search_mock(query=query, pageno=2)
+        self.store.call(self.store.plugins, 'post_search', request, search)
+        self.assertNotIn('user-agent', search.result_container.answers)
 
 
-class HashPluginTest(SearxTestCase):  # pylint: disable=missing-class-docstring
-    def test_PluginStore_init(self):
-        store = plugins.PluginStore()
+class PluginHashTest(SearxTestCase):  # pylint: disable=missing-class-docstring
+    def setUp(self):
```

Comment:
Same here - the setup is shared for all tests, and I moved the assert to its own test.